MarkBind Template for Student Portfolio

jingting1412 opened the pull request:
**What is the purpose of this pull request?**

- [ ] Documentation update
- [ ] Bug fix
- [x] Feature addition or enhancement
- [ ] Code maintenance
- [ ] DevOps
- [ ] Improve developer experience
- [ ] Others, please explain:

<!--
  If this pull request is addressing an issue, link to the issue: #xxx

  If this pull request completely addresses an issue, use one of the closing keywords: "Fixes #xxx" or "Resolves #xxx"
  https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

  Otherwise, elaborate further on the rationale of this pull request as needed
-->

**Overview of changes:**
Added a template portfolio website for users based on the existing default template generated when running `markbind init ---template` and [existing portfolio themes available on jekyll](https://jekyll-themes.com/category/portfolio). 

#2385 

**Anything you'd like to highlight/discuss:**

**Testing instructions:**
run `markbind init --template portfolio` and see if contents are loaded

<!--
  Any special testing instructions **not including** our automated tests.
-->

**Proposed commit message: (wrap lines at 72 characters)**
Add portfolio template for init command

<!--
  See this link for more info on how to write a good commit message:
  https://se-education.org/guides/conventions/git.html

  As best as possible, write a succinct commit title in 50 characters

|---------This is the width of 50 chars----------|
|-----------This is the width of 72 chars for your reference-----------|
-->

---

**Checklist:** :ballot_box_with_check:

<!-- Leave non-applicable items unchecked -->

- [x] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [x] Linked all related issues
- [x] No unrelated changes <!-- It's tempting, but increases the reviewer's work, and really pollutes the commit history =( -->

<!--
  We'll try our best to get to your PR within a week.
  If we haven't gotten to it then, or if your pull request resolves an urgent item, feel free to give us a ping!
-->

---

jingting1412 commented on packages/core/template/portfolio/_markbind/variables.md:
```diff
@@ -0,0 +1,4 @@
+<variable name="example">
+To inject this HTML segment in your MarkBind files, use {{ example }} where you want to place it.
+More generally, surround the segment's id with double curly braces.
+</variable>
```
Hello! Sorry for the late response! I've implemented the changes as suggested.

jingting1412 commented on docs/userGuide/templates.md:
```diff
@@ -31,6 +31,8 @@ Name    | Template key | Description | Quick Deploy
 ----    | -------      | ----------- | ------------
 Default | `default`    | Default template if `--template` is unspecified. Includes core features such as site and page navigation for a more convenient quick start. | <a href="https://app.netlify.com/start/deploy?repository=https://github.com/MarkBind/init-typical-netlify"><img src="https://www.netlify.com/img/deploy/button.svg" /></a>
 Minimal | `minimal` | Minimalistic template that gets you started quickly. | <a href="https://app.netlify.com/start/deploy?repository=https://github.com/MarkBind/init-minimal-netlify"><img src="https://www.netlify.com/img/deploy/button.svg" /></a>
+Portfolio | `portfolio` | Portfolio template to quickly build your portfolio. | <a href="https://heartfelt-faun-d2a60d.netlify.app"><img src="https://www.netlify.com/img/deploy/button.svg" /></a>
```
Yep for sure!

yucheng11122017 commented on packages/core/template/portfolio/contents/experience.md:
```diff
@@ -0,0 +1,9 @@
+
+# Experience
+
+> This is a placeholder page for your work experiences
```
Might be good to put the examples on using thumbnails and icons you have on the home page here instead!

jingting1412 replied:
Thanks for the idea! I think placing the thumbnails and icons on the homepage should remain though because many of the examples have such thumbnails/icons on the homepage to give a quick overview of their skills. I think if we were to move the thumbnails/icons from the homepage, maybe I could move to the skills page instead?

yucheng11122017 reviewed (commented):
Hi @jingting1412 thanks for the changes! 
One comment on moving the icons and thumbnails

yucheng11122017 reviewed (approved):
LGTM! Thanks for the work @jingting1412

yucheng11122017 commented:
Sorry @jingting1412, one more thing! 
Could you add functional test for the template?

jingting1412 commented:
> Sorry @jingting1412, one more thing! Could you add functional test for the template?

Yep added! Sorry for not noticing earlier

yucheng11122017 reviewed (approved):
LGTM! Will wait for more comments before merging :) 
@kaixin-hc @EltonGohJH @itsyme

lhw-1 commented:
Hi, thanks for the PR @jingting1412! I've taken a look at the template, and I just have a few broad comments / suggestions that has to do with the UI / UX rather than the implementation itself:

- For a student portfolio, I would imagine that a sleek look would be preferred instead of too much clutter (though we could assume that this would be their first ever portfolio that they are setting up). 
  - I think we can reduce the clutter in the top navbar to only contain the GitHub and the LinkedIn logo - the search function is unlikely to be used, and the topics on the sidebar should be sufficient for navigation, since we aren't expecting a particularly lengthy site.
  - Breadcrumbs may not be needed for the same reason - the site is unlikely to be too lengthy. 
- I think we can include "Skills" and "Experience" as topics on the main page rather than as a separate page?
  - "Skills" can be the set of tech that the student has used previously, so I'm thinking of maybe having three [Panels](https://markbind.org/userGuide/components/presentation.html#panels) titled "Languages", "Frameworks & Tools", and "Others" respectively. And within those panels, maybe we can include examples for these. The two portfolio links that @yucheng11122017 posted has some possible examples.
  - It would be nice to present "Experience" as a simple timeline similar to how LinkedIn presents it, so [Boxes](https://markbind.org/userGuide/components/presentation.html#boxes) might be suitable here? It might be also useful to make use of the vertical line created by a [Blockquote](https://markbind.org/userGuide/formattingContents.html#blockquotes) here.
- The main container on the main page can be used to display: Name, short description, and profile image. Basically, embed the "Hello there! ..." section's current content into the black container instead of repeating the same-ish content twice.
- We could add some styling for the individual projects pages as well, similar to the layouts used by the two aforementioned portfolio links!
- The "Navigating this site" section will need to be updated - and maybe we can also include some information indicating that this portfolio is a simplified version (if we do decide to remove some navbar features, etc), and if the student wants access to more features, they can use other templates instead.

These are suggestions rather than directions, and feel free to counter these if you feel that they are not preferable :+1:

jingting1412 commented:
Hello! Thank you @lhw-1 , @kaixin-hc for the feedback. I've revamped the look of the portfolio template so that it would look more sleek. All information (skills, experience, etc.) is just displayed on 1 page now and I hope that it would look more like the examples given.

One issue I'm facing is whether to add a section that introduces new users to the site, and to MarkBind features overall. I have added a small tip on the homepage that links users to the user guide, but I'm wondering if I should add a small section on introducing the site and give some tips about what features they might find useful (similar to the "Navigating this page" section on the current default template) to help first-time users editing the page.

kaixin-hc commented on packages/core/template/portfolio/index.md:
```diff
@@ -0,0 +1,43 @@
+<frontmatter>
+  title: Home Page
```
Keep in mind the use of frontmatter :https://markbind.org/userGuide/tweakingThePageStructure.html#frontmatter and omitFrontmatter 

(btw - if you're looking for more issues to pick up, you can consider adding to the discussion here: https://github.com/MarkBind/markbind/issues/2295 )

Anyway, because of the use of includes in this page, the page title is not Home Page but that of the last included file. Could you go through and remove irrelevant frontmatter? Might be worthwhile to also include in a code comment the purpose of frontmatter on this page

kaixin-hc commented on packages/core/template/portfolio/contents/project2.md:
```diff
@@ -0,0 +1,23 @@
+
+<br>
+
+### Project 2
```
Feels superfluous to have this title if all the content included is in the card! Having this also makes searching pretty tricky... Just type project into the search bar and you'll see what I mean. The duplication of titles on search is outside the scope of this PR though

kaixin-hc commented on packages/core/template/portfolio/contents/project2.md:
```diff
@@ -0,0 +1,23 @@
+  <div class="card-body">
+    <h5 class="card-title">Short description of project</h5>
+  </div>
+  <pic src="../contents/assests/MarkbindExample.png" width="500" alt="project-display"/>
```
What if we switched this from a pic to a youtube embed? Just so the projects look distinctly different. You can also replace the example text underneath to include info about embeds, and the example text under the first text to include links to pic + annotations

kaixin-hc commented on packages/core/template/portfolio/contents/experience.md:
```diff
@@ -0,0 +1,29 @@
+# Experience
+
+<panel type="seamless" header="Experience 1" no-close minimized>
+  <md>**Description of experience (Name of role, duration, etc.)**
```
maybe you could have a [badge](https://markbind.org/userGuide/components/presentation.html#tabs) with the date since normally you would list expereinces in chronological order

yucheng11122017 replied:
This could be even better with a full blown example of the experience 
I was thinking like 
```
**Company**: Full stack engineer
From {date badge} to {date badge}

Technical skills: 
- ....
```

What do you think

kaixin-hc commented on packages/core/template/portfolio/contents/project1.md:
```diff
@@ -0,0 +1,23 @@
+    {{typescript}}
+  </div>
+  <div class="card-body">
+    <a href="#" class="btn btn-primary">Card link</a>
```
I was confused as to what this should lead to. maybe instead of "Card Link" you can go with "Some External Link Here" or similar so its more clear its sample text?

kaixin-hc commented on packages/core/template/portfolio/contents/project1.md:
```diff
@@ -0,0 +1,23 @@
+<div class="card mb-3">
+  <h3 class="card-header">Name of project</h3>
+  <div class="card-body">
+    <h5 class="card-title">Short description of project</h5>
```
Don't think this needs to be a header. It doesn't look like a header either haha

kaixin-hc commented on packages/core/template/portfolio/contents/project2.md:
```diff
@@ -0,0 +1,23 @@
+  title: Project 2
+</frontmatter>
+
+<br>
```
Deleting this linebreak and the one in project 2 does not seem to have any effect, wld suggest removing them

kaixin-hc commented on packages/core/template/portfolio/_markbind/variables.json:
```diff
@@ -0,0 +1,3 @@
+{
+    "jsonVariableExample": "Your variables can be defined here as well"
```
Not used? Maybe remove?

kaixin-hc commented on packages/core/template/portfolio/404.md:
```diff
@@ -0,0 +1,7 @@
+<frontmatter>
+  title: Page not found
+  layout: 404.md
```
Is this accessible at any point? HAHA

kaixin-hc commented on packages/core/template/portfolio/contents/skills.md:
```diff
@@ -0,0 +1,26 @@
+    {{ typescript }}
+    {{ vue }}
+  </panel>
+  <br/>
```
I think you don't need a linebreak here, it's fine for it to all be on the same line?

kaixin-hc commented on packages/core/template/portfolio/contents/skills.md:
```diff
@@ -0,0 +1,26 @@
+  <panel header="Example using icons" minimized>
+    <box type="warning">
+      Some icons are not available in the libraries MarkBind supports! Please ensure that 
+      the desired icons are available beofre choosing this option. For more information regarding
```
```suggestion
      the desired icons are available before choosing this option. For more information regarding
```

kaixin-hc commented on packages/core/template/portfolio/index.md:
```diff
@@ -0,0 +1,43 @@
+  <md>[add relevant links](https://markbind.org/userGuide/formattingContents.html#links)</md>.
+</p>
+
+<img src='./contents/assests/default_profile_pic.png' alt='default-profile-pic'/>
```
more minor but i kinda feel like since this is left aligned the right side looks real empty?

kaixin-hc reviewed (commented):
Thanks for the revisions! I think it looks much better. Maybe @MarkBind/active-devs can chip in again?

Also, there's a typo in the folder `assests` - can you rename the folder? thanks!

Responding to your question - I think the tip is good but feel free to add more detail if you think it will help!

lhw-1 commented:
> One issue I'm facing is whether to add a section that introduces new users to the site, and to MarkBind features overall. I have added a small tip on the homepage that links users to the user guide, but I'm wondering if I should add a small section on introducing the site and give some tips about what features they might find useful (similar to the "Navigating this page" section on the current default template) to help first-time users editing the page.

I think introducing markbind features shouldn't be a role for the portfolio page, so your current approach should be sufficient - having said that, if you feel that there are specific markbind features that might specifically be useful for the portfolio page, you can add more of them in as tips 👍

yucheng11122017 reviewed (commented):
Hi @jingting1412 thanks for the work! 

Some minor nits and other comments. 
Do check out the documentation on site.json to be more familiar with how the site should be configured on markbind

yucheng11122017 commented:
Hi @jingting1412, 

Reminder about the changes for experiences and also to add the functional test!

ckcherry23 commented on docs/userGuide/templates.md:
```diff
@@ -31,6 +31,8 @@ Name    | Template key | Description | Quick Deploy
 ----    | -------      | ----------- | ------------
 Default | `default`    | Default template if `--template` is unspecified. Includes core features such as site and page navigation for a more convenient quick start. | <a href="https://app.netlify.com/start/deploy?repository=https://github.com/MarkBind/init-typical-netlify"><img src="https://www.netlify.com/img/deploy/button.svg" /></a>
 Minimal | `minimal` | Minimalistic template that gets you started quickly. | <a href="https://app.netlify.com/start/deploy?repository=https://github.com/MarkBind/init-minimal-netlify"><img src="https://www.netlify.com/img/deploy/button.svg" /></a>
+Portfolio | `portfolio` | Portfolio template to quickly build your portfolio. | <a href=""><img src="https://www.netlify.com/img/deploy/button.svg" /></a>
```
Hey Markbind team! I've been following this issue for quite a while but never got to see a preview of the Portfolio template. Maybe that is something you can consider adding to the documentation, perhaps a screenshot?

kaixin-hc replied:
Good point - it shouldn't be too hard to add a modal with image examples 👀 heres a short video i took when i was reviewing if you'd like to add your comments as well


https://github.com/MarkBind/markbind/assets/59471320/ffd799d2-1d0a-42be-b8a7-6b15493a1511

ckcherry23 replied:
Thanks! I took a short look during the lecture today too, and it looks clean! 

Not sure if this is possible on Markbind, but I think some of these expandable cards could be open by default? For example, the skills section, and maybe a few selected projects to emphasize the best ones.

I think the intro section ("My name is...") could be redesigned slightly if it is within the scope of this PR. For example, center the text, add a one-line intro, add call-to-action button(s) - for example, 'View Resume', or 'About Me', or the icon buttons for GitHub and LinkedIn can be added here

Just a few design suggestions, feel free to ignore

jingting1412 replied:
Hi Charisma! Thank you for your feedback. I think I can definitely change the skills section so the panels are open by default. For the projects section, each project is shown on an individual card so they are already open by default. I looked at your website again and I agree with the "featured projects" and "other projects". Maybe the current big cards can be for featured projects, and I can add a grid of smaller cards underneath that to show other projects.

I think the "About me" section looks better with the text aligned to the left? It looks neater in my opinion. 😓 
Your comment on the github/linkedin/resume links makes sense! I can def move it to the "About me" section. 

Also sorry about not having a netlify link previously, I've since set it up here: https://portfolio-template-markbind.netlify.app/

damithc commented:
@jingting1412 and others involved in  this PR: I suggest you keep CS2103/T students as the primary target of this feature. 
* As they are going for internships in near future, they might find setting up a portfolio like this quite useful
* A significant portion of them are not entirely new to MarkBind, as they have the option to use MarkBind for the tP (about 40% do)

Implications/suggestions:
* Optimize for NUS SoC students. Some placeholders can use NUS/SoC/CS so that students can use them even without modifying.
* Make it easy for them to plug in their iP and tP into the portfolio
* We might also want to provide some guidance on (and optimize the template for) hosting their portfolio page on GitHub pages so that `http://username.github.io` goes straight to their portfolio page. 

@jingting1412 can you host the site generated on GitHub pages and share the link here so that I (and others) can look at the final outcome and give feedback?

ckcherry23 commented:
> * Make it easy for them to plug in their iP and tP into the portfolio

Suggestion: If MarkBind supports iframes, you could consider [embedding RepoSense charts](https://reposense.org/ug/sharingReports.html#embeddable-widgets) for iP/tP into the portfolio template 👀

kaixin-hc commented:
@damithc thanks for the great suggestions! @jingting1412 has already deployed it here if you want to take a look: https://portfolio-template-markbind.netlify.app/ - though not on GitHub pages.

I'm personally not sure about the NUS/SoC customization - when its more clearly a placeholder, they're more likely to remember to change it, but if it looks close to "finished" you might forget to change it. But definitely think we can make it a _computer science_ portfolio page - especially in things like examples.

I second @ckcherry23 's suggestion to find a nice place to incorporate GitHub stats / reposense stats. It's a cool way to point at other NUS OSS projects.

jingting1412 commented:
@damithc thank you for the insight! Can I clarify if the approach means creating 2 templates, one generic, and the other one a full fleshed out sample of the first template for 2103 students? I think it’ll be helpful for 2103 students, but personally I have some similar reservations as @kaixin-hc in terms of using placeholders if we were to create a full sample. 

I think I can create some examples (e.g an example iP showcase) and guides (e.g guide to deploy on github) on the current template to give 2103 students a starting point on their portfolio while showcasing MarkBind’s features to other CS students.

damithc commented:
> @damithc thank you for the insight! Can I clarify if the approach means creating 2 templates, one generic, and the other one a full fleshed out sample of the first template for 2103 students? I think it’ll be helpful for 2103 students, but personally I have some similar reservations as @kaixin-hc in terms of using placeholders if we were to create a full sample.
> 
> I think I can create some examples (e.g an example iP showcase) and guides (e.g guide to deploy on github) on the current template to give 2103 students a starting point on their portfolio while showcasing MarkBind’s features to other CS students.

Yes, two, but I wouldn't call the second one a template. It's just a sample site that others can fork and tweak. Should not affect the MarkBind code at all (other than [here](https://markbind.org/showcase.html)), as it is a separate site altogether.

In the sample, we can write it as if John Doe is from NUS CS. For example, the university name can be NUS (there is no point putting a dummy placeholder for that and expecting everyone to replace the dummy value with NUS). Other things that are not so common can be left as placeholders.

jingting1412 commented:
Hello! I've implemented some of the suggestions discussed (embedding reposense, giving a guide on deploying to github pages, etc.) as well as adding tests. Any comments on how to improve this is greatly appreciated. The netlify link is here: https://portfolio-template-markbind.netlify.app/

kaixin-hc commented:
Thanks! I think this is good - but can we change this box? It's a bit hard to read with the code blocks...

<img width="1000" alt="Screenshot 2024-03-21 at 2 23 09 AM" src="https://github.com/MarkBind/markbind/assets/59471320/5f2967dc-a05f-49ba-8d28-2f6a743c424d">

kaixin-hc commented:
Oh! And can you check the reposense again? I think it might not be the widget that you display on the page

jingting1412 commented:
Hi! I've implemented the changes and fixed some issues to make the page responsive. The link is here again: https://portfolio-template-markbind.netlify.app/

kaixin-hc commented on docs/userGuide/templates.md:
```diff
@@ -31,6 +31,8 @@ Name    | Template key | Description | Quick Deploy
 ----    | -------      | ----------- | ------------
 Default | `default`    | Default template if `--template` is unspecified. Includes core features such as site and page navigation for a more convenient quick start. | <a href="https://app.netlify.com/start/deploy?repository=https://github.com/MarkBind/init-typical-netlify"><img src="https://www.netlify.com/img/deploy/button.svg" /></a>
 Minimal | `minimal` | Minimalistic template that gets you started quickly. | <a href="https://app.netlify.com/start/deploy?repository=https://github.com/MarkBind/init-minimal-netlify"><img src="https://www.netlify.com/img/deploy/button.svg" /></a>
+Portfolio | `portfolio` | Portfolio template to quickly build your portfolio. | <a href=""><img src="https://www.netlify.com/img/deploy/button.svg" /></a>
```
```suggestion
Portfolio | `portfolio` | Portfolio template to quickly build your portfolio. | <a href="">Quick deploy link to come!</a>
```

kaixin-hc reviewed (approved):
One more nit - the link shouldn't be blank and yet look the same as the other working links. I'm OK to merge this first though and make a separate issue for improving this section of the documentation to have demo/modals and to fix this button. 

@damithc @yucheng11122017 @EltonGohJH @itsyme do raise up any other concerns you all have!

yucheng11122017 commented on packages/core/template/portfolio/contents/projects.md:
```diff
@@ -0,0 +1,61 @@
+        <div class="card-body">
+            This is another project. 
+        </div>
+        <svg xmlns="http://www.w3.org/2000/svg" class="d-block user-select-none" width="100%" height="200" aria-label="Placeholder: Image cap" focusable="false" role="img" preserveAspectRatio="xMidYMid slice" viewBox="0 0 318 180" style="font-size:1.125rem;text-anchor:middle"><rect width="100%" height="100%" fill="#868e96"></rect><text x="50%" y="50%" fill="#dee2e6" dy=".3em">Image cap</text></sv ...
```
Could we change this to something more interesting? Maybe even the markbind logo? This also seems to be a good place to show off markbind's image and pic component (either or)

yucheng11122017 replied:
Same for the other image caps!

jingting1412 replied:
Hi for sure! I've replaced the svg with an markbind image

yucheng11122017 commented on packages/core/template/portfolio/_markbind/layouts/default.md:
```diff
@@ -0,0 +1,34 @@
+    <li><a href="{{baseUrl}}/index.html#skills" class="nav-link">Skills</a></li>
+    <li><a href="{{baseUrl}}/index.html#experience" class="nav-link">Experience</a></li>
+    <dropdown header="Projects" class="nav-link">
+      <li><a href="{{baseUrl}}/index.html#project-1" class="dropdown-item">Project 1</a></li>
```
Broken link

jingting1412 replied:
Good catch! I've fixed it in the newest version

yucheng11122017 commented on packages/core/template/portfolio/contents/experience.md:
```diff
@@ -0,0 +1,25 @@
+
+<panel type="seamless" no-close minimized>
+  <div slot="header">
+    <md>**Software Engineering Intern @ [NUS](https://nus.edu.sg/)** | May 2024 - Aug 2024</md>
```
Hmmm the links here don't work because the panel will always open instead

jingting1412 replied:
Hi I've changed the panels for this section to expanded so that its easier to click open the links.

jingting1412 replied:
Yep I've since changed the look of the experience section, inspired by the "Where I've worked" section on @ckcherry23 website. It now uses tabs and the company is linked in the header of the tab content. You can check it out here: https://portfolio-template-markbind.netlify.app/

yucheng11122017 commented on packages/core/template/portfolio/_markbind/layouts/default.md:
```diff
@@ -0,0 +1,34 @@
+    <li><a href="{{baseUrl}}/index.html#experience" class="nav-link">Experience</a></li>
+    <dropdown header="Projects" class="nav-link">
+      <li><a href="{{baseUrl}}/index.html#project-1" class="dropdown-item">Project 1</a></li>
+      <li><a href="{{baseUrl}}/index.html#project-2" class="dropdown-item">Project 2</a></li>
```
Could add the other projects here also?

kaixin-hc replied:
As in, add "Other Projects" - not indiv links to all the other projects

yucheng11122017 reviewed (commented):
Hi @jingting1412 thanks this is looking better and better! Some broken components

KevinEyo1 commented:
Hi @jingting1412, maybe you can add a email icon together with the github and linkedin icons for additional way to contact. Also can add a placeholder link to download the student's resume.

KevinEyo1 commented on packages/core/template/portfolio/index.md:
```diff
@@ -0,0 +1,62 @@
+</div>
+
+<box type="tip">
+  Want to change the theme? Explore more themes and other components offered by MarkBind in our <a href="https://markbind.org/userGuide/authoringContents.html" target="_blank">user guide here</a>
```
```suggestion
  Want to change the theme? Explore more themes and other components offered by MarkBind in our <a href="https://markbind.org/userGuide/authoringContents.html" target="_blank">user guide here</a>.
```

KevinEyo1 commented on packages/core/template/portfolio/index.md:
```diff
@@ -0,0 +1,62 @@
+</box>
+
+<box type="info" header="##### Deploying this site">
+    <md>MarkBind makes it easy to deploy this site to Github Pages using the `markbind deploy` command. To allow `http://[username].github.io` to go straight to this portfolio, it is best to set your repository name as `[username].github.io` Read more about [deploying the site here](https://markbind.org/userGuide/deployingTheSite.html#deploying-to-github-pages)</md>
```
```suggestion
    <md>MarkBind makes it easy to deploy this site to GitHub Pages using the `markbind deploy` command. To allow `http://[username].github.io` to go straight to this portfolio, it is best to set your repository name as `[username].github.io` Read more about [deploying the site here](https://markbind.org/userGuide/deployingTheSite.html#deploying-to-github-pages).</md>
```

KevinEyo1 commented on packages/core/template/portfolio/index.md:
```diff
@@ -0,0 +1,62 @@
+      <img src='./contents/assets/default_profile_pic.png' class="img-fluid" alt='default-profile-pic'/>
+    </div>
+    <div class="col-md-9 col-md-pull-3">
+      This is where you can add a picture of yourself and a brief introduction. This can be a good place to use <tooltip content="Add more information here" placement="top">tooltips</tooltip>, or <md>[add relevant links](https://markbind.org/userGuide/formattingContents.html#links)</md>
```
```suggestion
      This is where you can add a picture of yourself and a brief introduction. This can be a good place to use <tooltip content="Add more information here" placement="top">tooltips</tooltip>, or <md>[add relevant links](https://markbind.org/userGuide/formattingContents.html#links).</md>
```

jingting1412 replied:
Fixed all fullstop mistakes! Thank you for noticing this!

KevinEyo1 commented on packages/cli/test/functional/test_site_templates/test_portfolio/expected/index.html:
```diff
@@ -0,0 +1,260 @@
+              <p>Example using icons</p>
+            </template>
+            <box type="warning">
+              Some icons are not available in the libraries MarkBind supports! Please ensure that
```
```suggestion
              Some icons are not available in the libraries that MarkBind supports! Please ensure that
```

KevinEyo1 commented on packages/cli/test/functional/test_site_templates/test_portfolio/expected/index.html:
```diff
@@ -0,0 +1,260 @@
+            <box type="warning">
+              Some icons are not available in the libraries MarkBind supports! Please ensure that
+              the desired icons are available before choosing this option. For more information regarding
+              icons, please refer to <a href="https://markbind.org/userGuide/formattingContents.html#icons" target="_blank">this section of our User Guide</a></box>
```
```suggestion
              icons, please refer to <a href="https://markbind.org/userGuide/formattingContents.html#icons" target="_blank">this section of our User Guide</a>.</box>
```

KevinEyo1 commented on packages/cli/test/functional/test_site_templates/test_portfolio/expected/index.html:
```diff
@@ -0,0 +1,260 @@
+            <div class="card mb-3">
+              <h3 class="card-header" id="example-project-ip-for-cs2103">Example project: iP for CS2103<a class="fa fa-anchor" href="#example-project-ip-for-cs2103" onclick="event.stopPropagation()"></a></h3>
+              <div class="card-body">
+                <span>This would be a good place to add images and embeds showcasing your contributions to a project. In this case, an <code class="hljs inline no-lang" v-pre>&lt;iframe&gt;</code> containing the <a href="https://reposense.org/index.html">RepoSense</a> report of a student's contributions to their iP is embedded. For more information on embeds, refer to our user guide <a href="/con ...
```
Link here is broken

jingting1412 replied:
Yeah good catch! I've fixed it since, you can test it out here: https://portfolio-template-markbind.netlify.app/

yucheng11122017 replied:
Hi this is still broken

itsyme replied:
@jingting1412 I think you have to fix the broken link in the actual file instead of the expected file

jingting1412 replied:
Yep sorry 😔 it should be fixed now

KevinEyo1 commented on packages/cli/test/functional/test_site_templates/test_portfolio/expected/index.html:
```diff
@@ -0,0 +1,260 @@
+                <iframe src="https://nus-cs3281.github.io/2024-dashboard/#/widget/?search=Reposense&sort=groupTitle&sortWithin=title&timeframe=commit&mergegroup=&groupSelect=groupByAuthors&breakdown=false&since=2023-12-01&tabOpen=true&tabType=authorship&tabAuthor=asdfghjkxd&tabRepo=reposense%2Freposense%5Bmaster%5D&authorshipIsMergeGroup=false&authorshipFileTypes=java~md&authorshipIsBinaryFileTyp ...
+              </div>
+              <div class="card-body">
+                <p class="card-text">Some further elaboration of the project can be added here</p>
```
```suggestion
                <p class="card-text">Some further elaboration of the project can be added here.</p>
```

KevinEyo1 reviewed (commented):
Couple nits found.

jingting1412 commented:
Thanks everyone for the review! I've updated the site and it looks like this now: https://portfolio-template-markbind.netlify.app/

kaixin-hc commented on packages/core/template/portfolio/contents/projects.md:
```diff
@@ -0,0 +1,61 @@
+            {{ vue }}
+        </div>
+        <div class="card-body">
+            <a href="#" class="btn btn-primary">View on Github</a>
```
Suggest that this link should go to MarkBind github and the one below should go to MarkBind website

kaixin-hc replied:
Great! @jingting1412 the button next to this seems wrong tho? What is "post"

jingting1412 replied:
I was thinking that "Devpost" seems a bit inaccurate, since we're linking to documentation, but yeah "post" might be a little confusing too 😓. I'll change it back to Devpost

yucheng11122017 reviewed (approved):
LGTM except for the broken link

itsyme commented on packages/cli/test/functional/test_site_templates/test_portfolio/expected/index.md:
```diff
@@ -0,0 +1,70 @@
+        </a>
+      </span>
+    </div>
+    <a class="btn btn-primary" href="./contents/assets/ug_template.docx" download="UG-template-resume" role="button">Resume</a>
```
Perhaps using Download Resume could be more indicative to the viewer that it is a button to download as my expected intention when clicking the button is to be brought to a page showing the resume. Another way to go about this could be to exporting the docx file to pdf to allow it to be opened in a new tab instead of forcing a download. Imo using the pdf would be my preference

jingting1412 replied:
Yep makes sense! I've changed it to the pdf file, you can check it again here: https://portfolio-template-markbind.netlify.app/

itsyme reviewed (approved):
LGTM! Thanks for the research and effort put into creating this template!
